### PR TITLE
fix cryotubes from going to fusion level heat

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Added a max temperature to the cryotube of 4000 K, also if the temperature inside goes over 2000 K it will start to breakdown due to the heat

Also the tube extinguish the bodies inside of it

People discovered that a burning body inside of a tube will keep heating up the tube to infinity, let's not do that

## Why It's Good For The Game
No more infinite heat cryotubes

## Changelog
:cl:
fix: Cryotubes can no longer reach fusion level heat
add: Cryotubes breakdown if the heat inside is higher than 2000 K
add: Cryotubes extinguish the mob if on fire
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 1]
```diff
@@ -78,6 +80,15 @@
...
 	QDEL_NULL(radio)
 	QDEL_NULL(beaker)
+	///Take the turf the cryotube is on
+	var/turf/T = loc
+	///Take the air composition of the turf
+	var/datum/gas_mixture/env = T.return_air()
+	///Take the air composition inside the cryotube
+	var/datum/gas_mixture/air1 = airs[1]
+	env.merge(air1)
+	air_update_turf()
```
Comment (nemvar) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:90]:
```
	var/turf/T = get_turf(src)
	if(T)
		///Take the air composition of the turf
		var/datum/gas_mixture/env = T.return_air()
		///Take the air composition inside the cryotube
		var/datum/gas_mixture/air1 = airs[1]
		env.merge(air1)
		T.air_update_turf()
```
Alternatively use a ``is_turf(loc)``

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 2]
```diff
@@ -245,12 +256,15 @@
...
 
-			air1.temperature = max(air1.temperature - heat / air_heat_capacity, TCMB)
+			air1.temperature = clamp(air1.temperature - heat / air_heat_capacity, TCMB, MAX_TEMPERATURE)
 			mob_occupant.adjust_bodytemperature(heat / heat_capacity, TCMB)
 
 		air1.gases[/datum/gas/oxygen][MOLES] = max(0,air1.gases[/datum/gas/oxygen][MOLES] - 0.5 / efficiency) // Magically consume gas? Why not, we run on cryo magic.
 		air1.garbage_collect()
 
+		if(air1.temperature > 2000)
+			take_damage(clamp((air1.temperature)/300, 7, 15), BURN, 0, 0)
```
Comment (nemvar) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:266]:
```suggestion
			take_damage(clamp((air1.temperature)/300, 7, 15), BURN, sound_effect = FALSE)
```
Third argument is a string, fourth is a boolean.

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 3]
```diff
@@ -1,4 +1,6 @@
 #define CRYOMOBS 'icons/obj/cryo_mobs.dmi'
+///Max temperature allowed inside the cryotube, should break before reaching this heat
+#define MAX_TEMPERATURE 4000
```
Comment (nemvar) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:3]: #undef this at the buttom of the file.

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm (hunk 2 above)
Comment (Ghilker) [reply code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:266]: i'll just remove those then, leaving only the amount and the damage type

**Comment by Dorsisdwarf** [comment]
Cryotubes should probbably extinguish anything inside them. They're visibly filled with liquid, and strongly resemble the bacta tank from star wars.

**Comment by Ghilker** [comment]
now the cryotube extinguish the bodies inside

**Comment by Krysonism** [comment]
Rest in peace monkey engine

**Comment by Cyprex** [comment]
So? You can have fire underwater

**Comment by Ghilker** [comment]
Not normal fires tho

**Comment by ATH1909** [comment]
How does this interact with pyroxadone?

**Comment by Ghilker** [comment]
it doesnt affect pyro, because pyro max efficiency is around 500 K, well below the damaging heat cap

**Comment by MarioWizard119** [comment]
Don’t you also have to be on fire as well as 500K to be at max efficiency for pyrox?

**Pull request merged**